micrometer-api/src/main/java/io/micrometer/api/instrument/observation/TimerObservationHandler.java

```diff
@@ -0,0 +1,49 @@
+package io.micrometer.api.instrument.observation;
+
+import io.micrometer.api.instrument.MeterRegistry;
+import io.micrometer.api.instrument.Tag;
+import io.micrometer.api.instrument.Tags;
+import io.micrometer.api.instrument.Timer;
+
+/**
+ * Handler for {@link Timer.Sample}.
+ *
+ * @author Marcin Grzejszczak
+ * @since 2.0.0
+ */
+public class TimerObservationHandler implements ObservationHandler {
+
+    private final MeterRegistry meterRegistry;
+
+    public TimerObservationHandler(MeterRegistry meterRegistry) {
+        this.meterRegistry = meterRegistry;
+    }
+
+    @Override
+    public void onStart(Observation observation, Observation.Context context) {
+        Timer.Sample sample = Timer.start(meterRegistry);
+        context.put(Timer.Sample.class, sample);
+    }
+
+    @Override
+    public void onError(Observation observation, Observation.Context context, Throwable throwable) {
+        context.put(Throwable.class, throwable);
```

Comment:
If we could set an error field on the context in `Observation::error` we don't need to do this.